apps/designer/app/designer/features/sidebar-left/panels/components/components.tsx

```diff
@@ -1,90 +1,146 @@
-import { useEffect, useRef, type MouseEventHandler, useCallback } from "react";
-import { useDrag } from "react-dnd";
-import { getEmptyImage } from "react-dnd-html5-backend";
+import { type MouseEventHandler, useState } from "react";
+import { createPortal } from "react-dom";
 import {
   type Instance,
   type Publish,
   components,
 } from "@webstudio-is/react-sdk";
 import { Flex } from "@webstudio-is/design-system";
+import { useDrag } from "@webstudio-is/design-system";
 import { PlusIcon } from "@webstudio-is/icons";
-import { type DragData } from "~/shared/canvas-components";
 import { createInstance } from "~/shared/tree-utils";
 import type { TabName } from "../../types";
-import { CustomDragLayer } from "./custom-drag-layer";
 import { ComponentThumb } from "./component-thumb";
+import { useCanvasRect, useZoom } from "~/designer/shared/nano-states";
+import {
+  DragEndPayload,
+  DragMovePayload,
+  DragStartPayload,
+} from "~/canvas/shared/use-drag-drop";
+
+const componentNames = (
+  Object.keys(components) as Array<Instance["component"]>
+).filter((component) => components[component].isInlineOnly === false);
 
-type UseDraggableProps = {
+type DraggableThumbProps = {
+  onClick: MouseEventHandler<HTMLDivElement>;
   component: Instance["component"];
-  onDragChange: (isDragging: boolean) => void;
 };
 
-const useDraggable = ({ component, onDragChange }: UseDraggableProps) => {
-  const lastIsDragging = useRef<boolean | void>();
-  const [{ isDragging }, dragRef, preview] = useDrag(
-    () => ({
-      type: component,
-      collect: (monitor) => ({
-        isDragging: monitor.isDragging(),
-      }),
-    }),
-    []
+const DraggableThumb = ({ component, onClick }: DraggableThumbProps) => {
+  return (
+    <ComponentThumb
+      data-drag-component={component}
+      component={component}
+      onClick={onClick}
+    />
   );
-
-  useEffect(() => {
-    if (lastIsDragging.current !== undefined) onDragChange(isDragging);
-    lastIsDragging.current = isDragging;
-  }, [isDragging, onDragChange]);
-
-  useEffect(() => {
-    preview(getEmptyImage(), { captureDraggingState: true });
-  }, [preview]);
-
-  return dragRef;
 };
 
-type DraggableThumbProps = {
-  onClick: MouseEventHandler<HTMLDivElement>;
-} & UseDraggableProps;
-
-const DraggableThumb = ({
+const DragLayer = ({
   component,
-  onDragChange,
-  onClick,
-}: DraggableThumbProps) => {
-  const dragRef = useDraggable({ component, onDragChange });
-  return (
-    <ComponentThumb component={component} ref={dragRef} onClick={onClick} />
+  pointerPosition,
+}: {
+  component: Instance["component"];
+  pointerPosition: { x: number; y: number };
```

Comment:
would this be better?
```suggestion
  point: { x: number; y: number };
```